CMakeLists.txt

```diff
@@ -14,6 +14,15 @@ endforeach()
 
 project (benchmark)
 
+# Add a sensible build type default and warning because empty means no optimization and no debug info.
```

Comment:
this isn't the right approach as it forces a default. the correct approach is to guard the access to CMAKE_BUILD_TYPE in the pthread check with an `if(CMAKE_BUILD_TYPE)`.

Reply:
Interesting, this is the first time i'm seeing "it is okay to have no CMAKE_BUILD_TYPE".

Reply:
Well, cmake allows you to not specify CMAKE_BUILD_TYPE and, in my experience, it defaults to the equivalent of 'debug'. At least locally for me (and afai remember our build bots).